runtime: always use DIP_START and remove duplication of start packages in &rtp (#15632)
neovim/neovim: runtime/doc/repeat.txt

	pack/foo/start/foobar/syntax/some.vim
	pack/foo/opt/foodebug/plugin/debugger.vim

> When runtime files are searched for, first all paths in 'runtimepath' are

Reviewer: I'm thinking we will need to starting reference this in various places. It describes the Nvim "runtime path" but not the 'runtimepath' option.

```suggestion
*runtimepath*
When runtime files are searched for, first all paths in 'runtimepath' are
```

	pack/foo/start/foobar/syntax/some.vim
	pack/foo/opt/foodebug/plugin/debugger.vim

							*runtime-search-path*
When runtime files are searched for, first all paths in 'runtimepath' are
searched, then all "pack/*/start/*" dirs are searched. However, package entries

> are not visible in `:set rtp` or `echo &rtp`, as the final concatenated path
> would be too long and get truncated. To list all used directories, use

Reviewer: we don't really need to mention this reason, which is a pretty lame reason anyway (and doesn't capture our full reasoning?).

```suggestion
are not visible in `:set rtp` or `echo &rtp`. To list all used directories, use
```

> |nvim_list_runtime_paths()|. In addition |nvim_get_runtime_file()| can be used

Reviewer: active saves so many words and is clearer

```suggestion
|nvim_list_runtime_paths()|. Use |nvim_get_runtime_file()|
```

<	pack/foo/start/lib/autoload/foolib.vim >
		func foolib#getit()

> This works, because start packages will be used to look for autoload files,

Reviewer: ```suggestion
This works, because start packages will be searched for autoload files,
```
